File: src/main/java/com/woomulwoomul/woomulwoomulbackend/config/SentryConfig.java
package com.woomulwoomul.woomulwoomulbackend.config;

import com.woomulwoomul.woomulwoomulbackend.common.log.InterceptorLogging;
import io.sentry.Hint;
import io.sentry.SentryEvent;
import io.sentry.SentryOptions;
import io.sentry.protocol.SentryId;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.lang.Nullable;
import org.springframework.stereotype.Component;

@Component
public class SentryConfig implements SentryOptions.BeforeSendCallback {

    @Value("${api.name}")
    private String serverName;

    @Override
    public SentryEvent execute(SentryEvent event, @Nullable Hint hint) {
        event.setServerName(serverName);
        event.setEventId(new SentryId(InterceptorLogging.getRequestId()));
        return event;
    }
}

File: core/src/main/kotlin/com/woomulwoomul/core/config/SentryConfig.kt
package com.woomulwoomul.core.config

import com.woomulwoomul.core.common.log.InterceptorLogging
import io.sentry.Hint
import io.sentry.SentryEvent
import io.sentry.SentryOptions
import io.sentry.protocol.SentryId
import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Component

@Component
class SentryConfig(
    @Value("\${api.name}")
    private val serverName: String,
) : SentryOptions.BeforeSendCallback {

    override fun execute(event: SentryEvent, hint: Hint): SentryEvent? {
        event.serverName = serverName
        event.eventId = SentryId(InterceptorLogging().requestId)
        return event
    }
}